### PR TITLE
Add binary list dictionary

### PR DESCRIPTION
**Pull request by ebmifa**
## Description 
Add binary list dictionary

## Test plan 
https://github.com/MystenLabs/sui/actions/runs/9034639402/job/24827628678

### CODE REVIEW

File: binary-build-list.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[
+    "sui",
+    "sui-node",
+    "sui-tool",
+    "sui-faucet",
+    "sui-test-validator",
+    "sui-data-ingestion",
+    "sui-bridge",
+    "sui-bridge-cli",
+    "sui-graphql-rpc"
```
Comment (johnjmartin) [inline binary-build-list.json:10]: This doesn't match the list in `sui-release-binaries.yaml` btw

Comment (ebmifa) [reply binary-build-list.json:10]: Where did you find this file?

Comment (ebmifa) [reply binary-build-list.json:10]: Yes, cause that workflow is more for internal usage, and the original list here is more for public consumption.

Comment (ebmifa) [reply binary-build-list.json:10]: Like `stress`, `sui-proxy` , `sui-metric-checker`, `sui-analytics-indexer` and `sui-security-watchdog` binaries. But maybe we should package them all together for public consumption as well?

Comment (johnjmartin) [reply binary-build-list.json:10]: Nah i don't think those need to be added to the public binaries, but I do think we should have a second `binary-build-list.json` which `sui-release-binaries.yaml` uses

Comment (ebmifa) [reply binary-build-list.json:10]: Yep, that's coming next. I am just doing this on the sui public repo right now

**Review by johnjmartin — APPROVED** [review]

**Pull request merged**